File: fizzbuzz.r
#!/usr/bin/env Rscript
fizzbuzz <- function(n) {
  if ( length(n) == 1 )
    n <- seq_len(n)

  fizz <- !n %% 3
  buzz <- !n %% 5

  n[fizz & !buzz] <- "fizz"
  n[buzz & !fizz] <- "buzz"
  n[fizz &  buzz] <- "fizzbuzz"

  n
}

cat(fizzbuzz(100), sep = "\n")

File: fizzbuzz.scala
@main
def main(): Unit = {
  for (x <- 1 to 100)
    x match
      case x if x % 15 == 0 => println("fizzbuzz")
      case x if x % 5 == 0 => println("buzz")
      case x if x % 3 == 0 => println("fizz")
      case x => println(x)
}
